configure

```diff
@@ -672,41 +685,25 @@ def configure_node(o):
     o['variables']['node_target_type'] = 'static_library'
 
 
-def configure_libz(o):
-  o['variables']['node_shared_zlib'] = b(options.shared_zlib)
-
-  if options.shared_zlib:
-    o['libraries'] += ['-l%s' % options.shared_zlib_libname]
-  if options.shared_zlib_libpath:
-    o['libraries'] += ['-L%s' % options.shared_zlib_libpath]
-  if options.shared_zlib_includes:
-    o['include_dirs'] += [options.shared_zlib_includes]
-
-
-def configure_http_parser(o):
-    o['variables']['node_shared_http_parser'] = b(options.shared_http_parser)
-
-    if options.shared_http_parser:
-      o['libraries'] += ['-l%s' % options.shared_http_parser_libname]
-    if options.shared_http_parser_libpath:
-        o['libraries'] += ['-L%s' % options.shared_http_parser_libpath]
-    if options.shared_http_parser_includes:
-        o['include_dirs'] += [options.shared_http_parser_includes]
-
-
-def configure_libuv(o):
-  o['variables']['node_shared_libuv'] = b(options.shared_libuv)
-
-  if options.shared_libuv:
-    o['libraries'] += ['-l%s' % options.shared_libuv_libname]
-  if options.shared_libuv_libpath:
-    o['libraries'] += ['-L%s' % options.shared_libuv_libpath]
-  else:
-    o['variables']['uv_library'] = 'static_library'
-
-  if options.shared_libuv_includes:
-    o['include_dirs'] += [options.shared_libuv_includes]
-
+def configure_library(lib, output):
+  shared_lib = 'shared_%s' % lib
+  output['variables']['node_%s' % shared_lib] = b(getattr(options, shared_lib))
+
+  if getattr(options, shared_lib):
+    default_lib = format_libraries(getattr(options, '%s_libname' % shared_lib))
+    default_libpath = getattr(options, '%s_libpath' % shared_lib)
+    default_cflags = getattr(options, '%s_includes' % shared_lib)
+    
+    (pkg_libs, pkg_cflags) = pkg_config(lib)
+    libs = pkg_libs if pkg_libs else default_lib
+    cflags = pkg_cflags if pkg_cflags else default_cflags
+  
+    if libs:
+      output['libraries'] += libs.split()
+    if cflags:
+      output['include_dirs'] += cflags.split()
```

Comment:
Looking at it again, I don't include_dirs is the right field, it should be cflags or cflags_cc.  Probably the former.